### PR TITLE
Raise tresholds after which sampling and lightwood cache disabling are enabled

### PR DESCRIPTION
**Pull request by btseytlin**
Resolves: https://github.com/mindsdb/mindsdb_native/issues/68

Sampling for analysis now only enabled if the dataset is >= 30% of available RAM, same for disabling lightwood cache.
Sampling for training now only enabled if the dataset is >= 50% of available RAM.

### CODE REVIEW

**Review by maximlopin — APPROVED** [review]

File: mindsdb_native/libs/controllers/predictor.py
Comment (George3d6) [inline mindsdb_native/libs/controllers/predictor.py:26]: I agree that this should have been higher, but I think that, as per issue, this should actually be a function of the nr of rows up to a point (i.e. if we sample out of e.g. 500 rows, sampling might cause issues there). I will see how easy it is to get that logic in, do not merge for now, will merge when that is done.

**Review by George3d6 — CHANGES_REQUESTED** [review]

**Comment by George3d6** [comment]
Part of https://github.com/mindsdb/mindsdb_native/pull/77 with modifications, so closing this for now.

File: mindsdb_native/libs/controllers/predictor.py
Comment (btseytlin) [reply mindsdb_native/libs/controllers/predictor.py:26]: What if you have under 10k rows, but with 10k columns in each row?

**Pull request closed without merging**
